Selenium/selenium-dhammareg.py

```diff
@@ -0,0 +1,77 @@
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+import time
+from selenium.common.exceptions import NoSuchElementException  
+
+
+class DhammaReg():
+
+    def __init__(self, driver):
+        self.driver = driver
+        self.BASE_URL = "https://uscan.dhammareg.dhamma.org:7500/"
+        self.openDR()
+
+    def check_exists_by_xpath(self, xpath):
+        try:
+            self.driver.find_element_by_xpath(xpath)
+        except NoSuchElementException:
+            return False
+        return True
+
+    def openDR(self):
+        self.driver.get(self.BASE_URL)
+        self.driver.implicitly_wait(5)
+        uname = self.driver.find_element_by_xpath("//*[@id='LoginForm_username']")
+        uname.send_keys("nikhilj")
+        passw = self.driver.find_element_by_xpath("//*[@id='passwordField']")
+        passw.send_keys("beh@ppy0")
+        submit = self.driver.find_element_by_xpath("//*[@id='login-form']/div[4]/input")
+        submit.click()
+
+    def searchNumSat(self, expr):
+        numSat = self.driver.find_element_by_xpath("//input[@id='Students_NumberSat']")
+        numSat.clear()
+        numSat.send_keys(expr)
+        self.driver.find_element_by_xpath("//input[@name='yt0' and @value='Search']").click()
+        time.sleep(3)
+
+        if (self.check_exists_by_xpath(
+                "//div[@id='students-grid']/table//span[contains(text(), 'No results found.')]")):
+            return 0
+        else:
+            text = self.driver.find_element_by_xpath(
+                    "//div[@id='students-grid']/div[last() and contains(text(), 'Displaying')]").text
+            return text.split(' ')[3]
+
+    def advancedStudentSearch(self, city):
+        self.driver.get(self.BASE_URL + "students/Admin")
+        adv = self.driver.find_element_by_xpath("//a[contains(text(), 'Advanced Search')]")
+        adv.click()
+        cityInput = self.driver.find_element_by_xpath("//input[@id='Students_City']")
+        cityInput.clear()
+        cityInput.send_keys(city)
+        stateInput = self.driver.find_element_by_xpath("//input[@id='state']")
+        stateInput.clear()
+        stateInput.send_keys('CA')
+        
+        num_3 = self.searchNumSat(">2")
+        num_5 = self.searchNumSat(">4")
+        return (num_3, num_5)
+
+
+    def getProductCatLinks(self):        
+        nav_link = self.driver.find_element_by_xpath('id("popular_cat")')
+        return nav_link.find_elements_by_tag_name("a")
+
+
+def visitAllLinksSlow():
+    driver = webdriver.Firefox()
+    dr = DhammaReg(driver)
+    dr.openDR()
+    cities = ['san juan bautista', 'salinas', 'river oaks', 'prunedale', 'hollister', 'elkhorn']
+    for city in cities:
+        print(city, "  ", dr.advancedStudentSearch(city))
+
+    driver.close()
+
+visitAllLinksSlow()
```

Comment:
Does it matter if there is no end of line in python file? github shows me a warning about it.